src/hydro_system.hpp

```diff
@@ -422,6 +425,18 @@ AMREX_GPU_DEVICE AMREX_FORCE_INLINE auto HydroSystem<problem_t>::isStateValid(am
 	const amrex::Real rho = cons(i, j, k, density_index);
 	bool isDensityPositive = (rho > 0.);
 
+    bool isMassScalarPositive = true;
+	if (nmscalars_ > 0) {
```

Comment:
Can this be made `if constexpr` instead of just `if`?

Reply:
I dont know...

Reply:
It should be possible if `nmscalars_` is a `constexpr` variable.